userguide/advanceduse/reverse-tethering.rst

```diff
@@ -1,34 +1,76 @@
 This is useful if there is no available Wi-Fi connection or you don't have a data subscription on your Ubuntu Touch device.
 
-Prerequisite: A Ubuntu Touch device with "Developer mode" on, attached via USB to the computer connected to the Internet.
+Prerequisite: A Ubuntu Touch device with "Developer mode" on, attached via USB to the computer connected to the internet.
+
+Tested: 2023-01-02 with Ubuntu Touch 16.04 (OTA-24) and Ubuntu 22.04.
 
 Steps
 -----
 
-1. On the Ubuntu Touch device: ``android-gadget-service enable rndis``
+1. On the Ubuntu Touch device:
+
+  - Run: ``android-gadget-service enable rndis``
+
+  - Reboot the phone.
+
+2. On your computer:
 
-2. On your computer: get your RNDIS IP address e.g:``hostname -I``
+  - Get your RNDIS IP address:
 
-3. On the Ubuntu Touch device: 
+    Run: ``ip a``
 
-      add a gateway: ``sudo route add default gw YOUR_COMPUTER_RNDIS_IP``
+    Look for an interface named ``usb0`` in the output.
 
-      add nameservers: ``echo "nameserver 8.8.8.8" | sudo tee /etc/resolv.conf``
+    For example::
 
-      refresh the DNS table: ``resolvconf -u``
+        5: usb0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UNKNOWN group default qlen 1000
+           link/ether 76:eb:81:18:e0:40 brd ff:ff:ff:ff:ff:ff
+           inet 10.42.0.1/24 brd 10.42.0.255 scope global noprefixroute usb0
+              valid_lft forever preferred_lft forever
 
-4. On computer: 
+    Your IP is: ``10.42.0.1``
 
-      Turn on IP forwarding: ``echo 1 | sudo tee /proc/sys/net/ipv4/ip_forward``
+3. On the Ubuntu Touch device:
 
-      Apply NAT: ``sudo iptables -t nat -A POSTROUTING -s 10.0.0.0/8 -o eth0 -j MASQUERADE``
+  - Get your RNDIS interface: Run ``ip a``
+
+    For example::
   
+        31: rndis0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
+            link/ether b6:95:b6:08:4a:a8 brd ff:ff:ff:ff:ff:ff
+            inet 10.0.0.1/8 brd 10.255.255.255 scope global rndis0
+               valid_lft forever preferred_lft forever
+
+    Your interface is: ``rndis0``
+
+  - Change your IP address: ``sudo ifconfig rndis0 10.42.0.2``
+
+  - Add your computer as default gateway: ``sudo route add default gw 10.42.0.1``
 
+  - Add a nameserver of your choice: ``echo "nameserver 1.1.1.1" | sudo tee -a /etc/resolv.conf``
+
+  - Open Morph Browser and test your internet connection!
+
+4. On Ubuntu 22.04 NetworkManager already did this for you. If it's not working yet, on your computer:
+
+  - Turn on IP forwarding: ``echo 1 | sudo tee /proc/sys/net/ipv4/ip_forward``
+
+  - Apply NAT: ``sudo iptables -t nat -A POSTROUTING -s 10.42.0.0/8 -o eth0 -j MASQUERADE``
+
+    Make sure you replace ``eth0`` by the interface that connects your computer to the internet.
+
+Known Issues
+------------
+
+  - The Updates page in the System Settings displays: ``Connect to the Internet to check for updates.``
```

Comment:
Is this still true? Just asking. I do not know

Reply:
Yes, these are the issues I experienced. The RNDIS connection itself is pretty stable. Was connected to the phone over SSH for over 12 hours without an error. And the phone was online all the time as well.